shops/views.py

```diff
@@ -137,6 +138,37 @@ def post(self, request, *args, **kwargs):
                                 'pk': self.object.pk}))
 
 
+class ProductRemove(GroupPermissionMixin, ProductShopFromGroupMixin, View,
+                        GroupLateralMenuMixin):
+    """
+    Remove a product and redirect to the retrieve of the product.
+
+    :param kwargs['group_name']: name of the group used.
+    :param kwargs['pk']: pk of the product
+    :param self.perm_codename: codename of the permission checked.
+    """
+    template_name = 'shops/product_remove.html'
+    success_url = None
+    perm_codename = None
+
+    def get(self, request, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+        context['object'] = self.object
+        return render(request, self.template_name, context=context)
+
+    def post(self, request, *args, **kwargs):
+        # Set always to True, removing is non-reversible.
+        self.object.is_removed = True
+        self.object.save()
+
+        # Delete all category_product which use the product.
+        CategoryProduct.objects.filter(product=self.object).delete()
+
```

Comment:
Maybe you should also delete Inventory / Stock objects related to this product ?

Reply:
Hmm, they'll not be used anymore you're right. Though, I don't know if it's useful to delete them.

Reply:
If we completely remove the reversibility, it can free some space.

Particularly if the object is deleted after some wrong manipulation... Like it happened before.